### PR TITLE
Docs: Cleanup meta field docs

### LINKED ISSUE #10879: Docs: Update docs for mapping fields
With the changes in #8870, the mapping fields (`_all`, `_id` etc) are now very limited in how they can be configured, but the docs still reflect the old config options: 

http://www.elastic.co/guide/en/elasticsearch/reference/master/mapping-fields.html

### PR DESCRIPTION
**Pull request by rjernst**
Meta fields were locked down to not allow exotic options to the
underlying field types in #8143. This change fixes the docs
to no longer refer to the old settings.

closes #10879

### CODE REVIEW

File: docs/reference/mapping/fields/type-field.asciidoc [hunk 1]
```diff
@@ -1,31 +1,10 @@
...
-
-[source,js]
---------------------------------------------------
-{
-    "tweet" : {
-        "_type" : {"index" : "no"}
-    }
-}
---------------------------------------------------
+Note, even though the `_type` is not indexed, all the APIs still work
```
Comment (jpountz) [inline docs/reference/mapping/fields/type-field.asciidoc]: Hmm, but it IS indexed, right?

Comment (rjernst) [inline docs/reference/mapping/fields/type-field.asciidoc]: Yes! This was a mistake, I reworded the `_type` docs.

File: docs/reference/mapping/fields/field-names-field.asciidoc [hunk 2]
```diff
@@ -1,9 +1,7 @@
 [[mapping-field-names-field]]
 === `_field_names`
+added[1.3.0]
```
Comment (clintongormley) [inline docs/reference/mapping/fields/field-names-field.asciidoc]: No need for annotations from 1.x

File: docs/reference/mapping/fields/routing-field.asciidoc [hunk 3]
```diff
@@ -5,15 +5,6 @@ The routing field allows to control the `_routing` aspect when indexing
...
 [float]
-==== store / index
-
-The first thing the `_routing` mapping does is to store the routing
-value provided (`store` set to `true`) and index it (`index` set to
-`not_analyzed`). The reason why the routing is stored by default is so
-reindexing data will be possible if the routing value is completely
-external and not part of the docs.
-
-[float]
```
Comment (clintongormley) [inline docs/reference/mapping/fields/routing-field.asciidoc]: In the next para, delete "or derived from the doc".  

Also, need to check with @martijnvg, but I think you can delete "A delete operation will be broadcasted to all shards if no routing value is provided and `_routing` is required."

File: docs/reference/query-dsl/filters/ids-filter.asciidoc [hunk 4]
```diff
@@ -2,9 +2,7 @@
 === Ids Filter
 
 Filters documents that only have the provided ids. Note, this filter
-does not require the <<mapping-id-field,_id>>
-field to be indexed since it works using the
-<<mapping-uid-field,_uid>> field.
+uses the <<mapping-uid-field,_uid>> field.
```
Comment (clintongormley) [inline docs/reference/query-dsl/filters/ids-filter.asciidoc]: Add "If no `type` is specified, all types defined in the index mapping are tried." (or something like that)

File: docs/reference/query-dsl/filters/type-filter.asciidoc [hunk 5]
```diff
@@ -2,8 +2,7 @@
 === Type Filter
 
 Filters documents matching the provided document / mapping type. Note,
-this filter can work even when the `_type` field is not indexed (using
-the <<mapping-uid-field,_uid>> field).
+this filter uses the <<mapping-uid-field,_uid>> field.
```
Comment (clintongormley) [inline docs/reference/query-dsl/filters/type-filter.asciidoc]: Types are indexed, no? So surely it doesn't use the UID field?

File: docs/reference/query-dsl/queries/ids-query.asciidoc [hunk 6]
```diff
@@ -2,9 +2,7 @@
 === Ids Query
 
 Filters documents that only have the provided ids. Note, this filter
-does not require the <<mapping-id-field,_id>>
-field to be indexed since it works using the
-<<mapping-uid-field,_uid>> field.
+uses the <<mapping-uid-field,_uid>> field.
```
Comment (clintongormley) [inline docs/reference/query-dsl/queries/ids-query.asciidoc]: as per comment on the ids filter

**Comment by clintongormley** [comment]
As a side note, shouldn't we remove the ability to configure store/index on timestamp? http://www.elastic.co/guide/en/elasticsearch/reference/master/mapping-timestamp-field.html#_store_index_2

File: docs/reference/mapping/fields/routing-field.asciidoc [hunk 7]
```diff
@@ -5,15 +5,6 @@ The routing field allows to control the `_routing` aspect when indexing
 data and explicit routing control is required.
```
Comment (clintongormley) [inline docs/reference/mapping/fields/routing-field.asciidoc]: Add a note saying that `_routing` is stored but not indexed.

Comment (clintongormley) [inline docs/reference/mapping/fields/routing-field.asciidoc]: Actually looks like it is stored AND indexed

File: docs/reference/mapping/fields/routing-field.asciidoc (hunk 3 above)
Comment (rjernst) [inline docs/reference/mapping/fields/routing-field.asciidoc]: Looks like this was removed in #10136. I've removed that delete operation line as well.

File: docs/reference/query-dsl/filters/type-filter.asciidoc (hunk 5 above)
Comment (rjernst) [inline docs/reference/query-dsl/filters/type-filter.asciidoc]: This was just a goof on my part, removed.

File: docs/reference/query-dsl/queries/ids-query.asciidoc (hunk 6 above)
Comment (rjernst) [inline docs/reference/query-dsl/queries/ids-query.asciidoc]: `_id` is not stored or indexed. The ids query does use `_uid`.

**Comment by rjernst** [comment]
@clintongormley I pushed a new commit addressing your comments. For `_timestamp` we never locked it down because it is essentially a user field (the only way to add a NOW to a date field). I think we should continue looking at changing that separately.

**Pull request merged**